melodies_monet/driver.py

```diff
@@ -148,41 +148,76 @@ def open_obs(self, time_interval=None):
         """
         from glob import glob
         from numpy import sort
-        from . import tutorial
-
-        if self.file.startswith("example:"):
-            example_id = ":".join(s.strip() for s in self.file.split(":")[1:])
-            files = [tutorial.fetch_example(example_id)]
-        else:
-            files = sort(glob(self.file))
-
-        assert len(files) >= 1, "need at least one"
-
-        _, extension = os.path.splitext(files[0])
+ 
         try:
-            if extension in {'.nc', '.ncf', '.netcdf', '.nc4'}:
-                if len(files) > 1:
-                    self.obj = xr.open_mfdataset(files)
-                else:
-                    self.obj = xr.open_dataset(files[0])
-            elif extension in ['.ict', '.icarrt']:
-                assert len(files) == 1, "monetio.icarrt.add_data can only read one file"
-                self.obj = mio.icarrt.add_data(files[0])
-            else:
-                raise ValueError(f'extension {extension!r} currently unsupported')
-        except Exception as e:
-            print('something happened opening file:', e)
-            return
+            if os.path.isfile(self.file):
+                _, extension = os.path.splitext(self.file)
```

Comment:
This was something we all decided on early on in our code development, and ran past the main MELODIES MONET group. At that time we were also considering creating an open_aircraft_obs() section as well.

The open_obs() routine does not have functionality to test/flag whether surface or satellite data is being opened. Also, open_obs() has a more general way of opening files, whereas satellite obs require a different method for each satellite dataset being read. The satellite readers are structured more like opening and reading model files.

Consequently, we decided it was cleaner to have a whole different opening code section for satellites, rather than implement a huge external if statement.

Reply:
@rrbuchholz I understand and agree that it's easier/cleaner to have a whole different opening section for opening the satellite obs but here it seems like we are regressing the open_obs() functionality along with adding the open_sat_obs() functionality. Is that intentional or was the `observation.open_obs()` code in the satellite branch outdated and has been unintentionally propagated forward?

Reply:
@colin-harkins Ah I see, yes we are on the same page. We think the open_obs() was messed up in the merge and have addressed all your identified issues to align it with the main/develop branch.

Reply:
In the develop_satellite branch, observation.open_obs, the xr.open_dataset(self.file[0]) has been nested in an if os.path.isfile(self.file) block.  self.mask_and_scale() and self.filter_obs() have are also inside that block.  There is no else block with that conditional.  Not sure this if statement will work, or what was intended here ...

Reply:
Restored observation.open_obs to the develop branch version.